arcaflow_plugin_stressng/stressng_plugin.py

```diff
@@ -101,28 +97,10 @@ def stressng_run(
     system_info = stressng_yaml["system-info"]
     metrics = stressng_yaml["metrics"]
 
-    # allocate all stressor information with None in case they don't get called
-    cpuinfo_un = None
-    vminfo_un = None
-    mmapinfo_un = None
-    matrixinfo_un = None
-    mqinfo_un = None
-    hddinfo_un = None
-
     system_un = system_info_output_schema.unserialize(system_info)
-    for metric in metrics:
-        if metric["stressor"] == "cpu":
-            cpuinfo_un = cpu_output_schema.unserialize(metric)
-        if metric["stressor"] == "vm":
-            vminfo_un = vm_output_schema.unserialize(metric)
-        if metric["stressor"] == "mmap":
-            mmapinfo_un = mmap_output_schema.unserialize(metric)
-        if metric["stressor"] == "matrix":
-            matrixinfo_un = matrix_output_schema.unserialize(metric)
-        if metric["stressor"] == "mq":
-            mqinfo_un = mq_output_schema.unserialize(metric)
-        if metric["stressor"] == "hdd":
-            hddinfo_un = hdd_output_schema.unserialize(metric)
+    un = {
+        m["stressor"]: stressor_schemas[m["stressor"]].unserialize(m) for m in metrics
+    }
```

Comment:
Because now you don't have other info tied to the variable, there is now room for a better variable name than `un`.
In addition, if you're going to use a very python-specific way of producing a dict (I think), please comment it. This one line does a lot, and it's not immediately clear.

Reply:
Let me know if the comment needs more (or if you don't like my replacement for `un`).

Reply:
I was "un"enthused about "`un`" but decided to leave it "un"commented as I didn't really have a brilliant alternative suggestion. I guess `results` is a little better, although that's still pretty "un"specific, and I'm not that much less "un"enthused. But matching it up with your `stressor_schemas` we might try `stressor_results`, or `stressor_output`?

Reply:
I was good with `un` as shorthand for "unserialized output using stressor schemas"; we could certainly go with `stressor_results` or `stressor_output`, but I was thinking that expressions like
   ```go
   stressor_schemas[m["stressor"]].unserialize(m)
   ```
and
   ```go
   results.get(Stressors.HDD)
   ```
made it clear enough that there was a stressor involved that I didn't need to include it in the variable name.

Reply:
The new name seems at least better than `un`. Longer alternatives, if either of you would prefer them, include `unserialized_results` or `unserialized_stressor_results`.
The comment seems sufficient.

Reply:
The problem with adding the "unserialized_" prefix is that, at the point where the variable is read, we really don't care that the values were previously serialized or that they are unserialized now -- what we care about is that they are the results of the various stressor runs.

Longer names are not necessarily better.  While we do want to avoid names which are "too short" and names which are cryptic (really, `un` was not acceptable on its own -- I used it only in the context of what it was replacing, and because we still have `system_un`), having too much in a name leads to bloat and unnecessary cognitive load of various kinds.

A good rule of thumb is that a name's length should be proportional to the size of the scope over which it is visible.  For very small scopes (e.g., the dictionary "comprehension" that I added), single character names are perfectly acceptable and even good.  Most global variables, on the other hand, should have unpleasantly long and verbose names to discourage people from defining or using them.  😁

But, really, the goal should be for the code to read in a natural fashion.  So, having "stressor" appearing multiple times in the same expression is probably not what we want, and having "unserialized" occur in a context where we are neither unserializing nor serializing is probably counterproductive.  And, in this instance, the scope is only twenty lines, so a relatively small "mnemonic handle" will suffice.